pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -233,29 +251,49 @@ def __validate_init_configuration(self):
                     ' You can save the last checkpoint with ModelCheckpoint(save_top_k=None, monitor=None)'
                 )
 
-    def __init_ckpt_dir(self, filepath, save_top_k):
-        self._fs = get_filesystem(filepath if filepath is not None else "")
+    def __init_ckpt_dir(self, filepath, dirpath, filename, save_top_k):
+        if filepath:
+            if (dirpath or filename):
+                raise MisconfigurationException(
+                    f'You have set all three path/name inputs which are not feasible.'
+                    f' You have to choose either filepath={filepath} OR dirpath={dirpath}'
+                    f' and filename={filename} configuration.'
+                )
+
+            rank_zero_warn(
+                'Argument `filepath` is deprecated in v1.0 and will be removed in v1.2.'
+                ' Please use `dirpath` and `filename` instead.', DeprecationWarning
+            )
+
+            _fs = get_filesystem(filepath)
+
+            if _fs.isdir(filepath):
+                dirpath, filename = filepath, None
+            else:
+                if _fs.protocol == 'file':
+                    filepath = os.path.realpath(filepath)
+                dirpath, filename = os.path.split(filepath)
+
+        dirpath = str(dirpath) if dirpath else dirpath
+        self._fs = get_filesystem(dirpath or '')
```

Comment:
```suggestion
        self._fs = get_filesystem(str(dirpath) if dirpath else '')
```

Reply:
And with this suggestion, in line 295, you can have

```python
    self.dirpath = dirpath
```

Reply:
`self.dirpath = dirpath or None` will handle the case when `dirpath=''`